DQM/DTMonitorModule/interface/DTDataIntegrityTask.h

```diff
@@ -96,11 +105,13 @@ class DTDataIntegrityTask: public DQMEDAnalyzer {
   // Monitor Elements
   MonitorElement* nEventMonitor;
   // <histoType, <index , histo> >
-  std::map<std::string, std::map<int, MonitorElement*> > dduHistos;
+  std::map<std::string, std::map<int, MonitorElement*> > fedHistos;
   // <histoType, histo> >
-  std::map<std::string, std::map<int, MonitorElement*> > rosSHistos;
+  std::map<std::string, std::map<int, MonitorElement*> > summaryHistos;
   // <histoType, <index , histo> >
   std::map<std::string, std::map<int, MonitorElement*> > rosHistos;
+  // <histoType, <fed , index, histo> >
+  std::map<std::string, std::map<int, std::map< int, MonitorElement*> > > urosHistos;
```

Comment:
hi @jfernan2 - we'll need a different solution to this without the map of maps of maps. Perhaps  a more generic key  that could flatten the 4 existing maps of maps at the same time?

Reply:
hi @davidlange6 Thanks for your comments
I don't know of any other solution, there's no generic key to use as in the ROS case above, apart from reducing 1 level by creating 3:
std::map<std::string, std::map<int, MonitorElement*> > urosHistosFED1368
std::map<std::string, std::map<int, MonitorElement*> > urosHistosFED1369
std::map<std::string, std::map<int, MonitorElement*> > urosHistosFED1370
which in terms of memory I guess is the same and complicates the code. But I'm open to any suggestion.

I'm implementing now your other comments but, since this PR is needed for pre3 to have DT data in January, unless you have a strong objection I would beg you limit your comments to the non DQM modules since DQM can be fixed in other PR afterwards. We made the effort of including DQM changes in this same PR to ease the transition due to the DQM tests in the Jenkins checks but this is complicating the issue.
Thanks

Reply:
less - maps of maps are already bad.. maps^3 simply can not go.

reasonable solutions of course depend on the number of values of each key (roughly, or maximumly).  i understand the string key has 3 values (nice :( )  - how about the ints?

Reply:
This map is a variable array of up to 5 (string) x 3 feds (int) x 12 uROS (int) x ME
Can you give an example of what you are suggesting please?

Reply:
std::map<unsigned int, MonitorElement> 
with key just equal to stringEnum*1000 + (fed-minFED)#*100 + (uROS-minuROS)#  
(where the string values need to be turned into a set of ints  and fed, uROS are assumed to be contiguous values, which perhaps they are)